src/librustdoc/html/render.rs

```diff
@@ -143,6 +144,33 @@ impl Impl {
     }
 }
 
+#[derive(Debug)]
+pub struct IoError {
+    filename: String,
+    error: io::Error,
+}
+
+impl Error for IoError {
+    fn description(&self) -> &str {
+        self.error.description()
+    }
+}
+
+impl Display for IoError {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        write!(f, "\"{}\": {}", self.filename, self.error)
+    }
+}
+
+impl From<io::Error> for IoError {
```

Comment:
Right now the error message printed out would be:

```
"": <error text>
```

The leading `"": ` shouldn't happen if the file name is blank as the only context available is the error.